pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3837,6 +3838,24 @@ public void endReplaceSegments(String tableNameWithType, String segmentLineageEn
         tableNameWithType, segmentLineageEntryId);
   }
 
+  /**
+   * Internal method to initiate pageCache warmup for a table before the new refresh segments are available
+   * for querying.
+   * This method triggers a pageCache warmup operation on the server for the specified table and segments.
+   * For refresh tables, the `segmentsTo` list is particularly important as it contains the segments
+   * that need to be warmed up.
+   * Example:
+   * To warm up specific segments of the "salesData_OFFLINE" table:
+   *   - tableNameWithType: "salesData_OFFLINE"
+   *   - segmentsTo: ["segment1", "segment2", "segment3"]
+   *
+   * @param tableNameWithType The name and type of the table for which the pageCache warmup is triggered
+   * @param segmentsTo A list of segments that need to be warmed up before query availability
+   */
+  protected void preSegmentReplaceOperation(String tableNameWithType, List<String> segmentsTo) {
```

Comment:
Suggest renaming it to `preSegmentReplaceUpdateRouting()` to be more precise, and include `segmentsFrom` in case user wants to do certain things on segments to be replaced.

Reply:
Renamed and included `segmentsFrom` as well